server/sonar-web/src/main/js/app/components/nav/global/GlobalNavUser.js

```diff
@@ -19,21 +19,40 @@
  */
 // @flow
 import React from 'react';
-import { Link, withRouter } from 'react-router';
+import { sortBy } from 'lodash';
+import { Link } from 'react-router';
 import Avatar from '../../../../components/ui/Avatar';
+import OrganizationLink from '../../../../components/ui/OrganizationLink';
 import { translate } from '../../../../helpers/l10n';
 
-class GlobalNavUser extends React.PureComponent {
-  props: {
-    currentUser: {
-      email?: string,
-      name: string
-    },
-    location: Object,
-    router: { push: string => void }
-  };
+type Props = {
+  currentUser: {
+    email?: string,
+    isLoggedIn: boolean,
+    name: string
+  },
+  fetchMyOrganizations: () => Promise<*>,
+  location: Object,
+  organizations: Array<{ key: string, name: string }>,
+  router: { push: string => void }
+};
+
+export default class GlobalNavUser extends React.PureComponent {
+  props: Props;
+
+  componentDidMount() {
+    if (this.props.currentUser.isLoggedIn) {
+      this.props.fetchMyOrganizations();
```

Comment:
let's lazy load organizations, i.e. do the request only when a user opens the dropdown.